### PR TITLE
chore: make accessors public and reexport indexmap

### PR DESCRIPTION
**Pull request by broody**
Does it make sense to make the value accessors public? Currently, I'm duplicating this file in my own project to use them, but these are nice helpers I think the library could expose. Also, re-exporting the indexmap.

### CODE REVIEW

File: src/dynamic/value_accessor.rs [hunk 1]
```diff
@@ -160,7 +160,7 @@ impl<'a> ObjectAccessor<'a> {
 }
 
 /// A list accessor
-pub struct ListAccessor<'a>(pub(crate) &'a [Value]);
+pub struct ListAccessor<'a>(pub &'a [Value]);
```
Comment (sunli829) [inline src/dynamic/value_accessor.rs:163]: add `ListAccessor::as_slice()` method?

File: src/dynamic/value_accessor.rs [hunk 2]
```diff
@@ -106,7 +106,7 @@ impl<'a> ValueAccessor<'a> {
 }
 
 /// A object accessor
-pub struct ObjectAccessor<'a>(pub(crate) Cow<'a, IndexMap<Name, Value>>);
+pub struct ObjectAccessor<'a>(pub Cow<'a, IndexMap<Name, Value>>);
```
Comment (sunli829) [inline src/dynamic/value_accessor.rs:109]: add `ObjectAccessor::as_index_map()` method?

**Comment by broody** [comment]
@sunli829 lemme know if there's anything else I should add

File: src/dynamic/value_accessor.rs (hunk 1 above)
Comment (sunli829) [reply src/dynamic/value_accessor.rs:169]: keep `pub(crate)`

File: src/dynamic/value_accessor.rs [hunk 3]
```diff
@@ -106,7 +106,7 @@ impl<'a> ValueAccessor<'a> {
 }
 
 /// A object accessor
-pub struct ObjectAccessor<'a>(pub(crate) Cow<'a, IndexMap<Name, Value>>);
```
Comment (sunli829) [inline src/dynamic/value_accessor.rs:109]: keep `pub(crate)`

**Pull request merged**